### PR TITLE
Use the k8s sets for the code that is actually set of strings

### PR DESCRIPTION
**Pull request by vagababov**
/cc @mattmoor

### CODE REVIEW

File: apis/duck/v1alpha1/condition_set.go [hunk 1]
```diff
@@ -280,7 +280,7 @@ func (r conditionsImpl) MarkUnknown(t ConditionType, reason, messageFormat strin
 			// Double check that the happy condition is also false.
 			happy := r.GetCondition(r.happy)
 			if !happy.IsFalse() {
-				r.MarkFalse(r.happy, reason, messageFormat, messageA)
+				r.MarkFalse(r.happy, reason, messageFormat, messageA...)
```
Comment (vagababov) [inline apis/duck/v1alpha1/condition_set.go]: This change was required for compilation reasons, not sure how it was working before.

**Comment by vagababov** [comment]
As part of https://github.com/knative/serving/issues/3049

File: apis/duck/v1alpha1/condition_set.go (hunk 1 above)
Comment (mattmoor) [reply apis/duck/v1alpha1/condition_set.go]: 😲

**Pull request merged**